generate/generate.go

```diff
@@ -653,7 +654,7 @@ func (g *Generator) convertMessage(tspec *ast.TypeSpec) (*desc.DescriptorProto,
 		if len(field.Names) != 1 {
 			return nil, fmt.Errorf("need all fields to have one name")
 		}
-		fieldName := field.Names[0].Name
+		fieldName := snaker.CamelToSnake(field.Names[0].Name)
```

Comment:
hmm, not sure if I understand this. For example 

```
type Test1 struct {
  Field1 string `pb:"1" json:"field2_yes_not_a_typo"`
}
```

I would expect the generated .go file has

```
type Test1 struct {
  Field 1 string `... json:"field2_yes_not_a_typo,omitempty"`
}
```

That means we shouldn't need to do any case transformation.

I haven't really look at the surrounding codes, just this line, but this does need explanation.